### PR TITLE
removed AutoValue from geoJson

### PR DESCRIPTION
**Pull request by osana**
address #710

### CODE REVIEW

File: services-geojson/src/main/java/com/mapbox/geojson/GeometryCollection.java [hunk 1]
```diff
@@ -198,6 +229,159 @@ public String toJson() {
...
+    @Override
+    @SuppressWarnings("unchecked")
+    public void write(JsonWriter jsonWriter, GeometryCollection object) throws IOException {
+      if (object == null) {
+        jsonWriter.nullValue();
+        return;
+      }
+      jsonWriter.beginObject();
+      jsonWriter.name("type");
+      if (object.type() == null) {
```
Comment (LukasPaczos) [inline services-geojson/src/main/java/com/mapbox/geojson/GeometryCollection.java]: Same as above.

File: services-geojson/src/main/java/com/mapbox/geojson/GeometryCollection.java [hunk 2]
```diff
@@ -198,6 +229,159 @@ public String toJson() {
...
+                  (TypeAdapter<List<Geometry>>) gson.getAdapter(typeToken);
+          this.listGeometryAdapter = listGeometryAdapter;
+        }
+        listGeometryAdapter.write(jsonWriter, object.geometries());
+      }
+      jsonWriter.endObject();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
```
Comment (LukasPaczos) [inline services-geojson/src/main/java/com/mapbox/geojson/GeometryCollection.java]: Same as above.

File: services-geojson/src/main/java/com/mapbox/geojson/Polygon.java [hunk 3]
```diff
@@ -374,4 +390,70 @@ private static boolean isLinearRing(LineString lineString) {
...
+   * @since 4.4.0
+   */
+  static final class GsonTypeAdapter extends BaseGeometryTypeAdapter<Polygon, List<List<Point>>> {
+
+    GsonTypeAdapter(Gson gson) {
+      super(gson, new ListOfListOfPointCoordinatesTypeAdapter());
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
```
Comment (LukasPaczos) [inline services-geojson/src/main/java/com/mapbox/geojson/Polygon.java]: Same as above.

File: services-geojson/src/main/java/com/mapbox/geojson/Polygon.java [hunk 4]
```diff
@@ -374,4 +390,70 @@ private static boolean isLinearRing(LineString lineString) {
...
+    public void write(JsonWriter jsonWriter, Polygon object) throws IOException {
+      writeCoordinateContainer(jsonWriter, object);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
```
Comment (LukasPaczos) [inline services-geojson/src/main/java/com/mapbox/geojson/Polygon.java]: Same as above.

File: services-geojson/src/main/java/com/mapbox/geojson/gson/BoundingBoxTypeAdapter.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import com.mapbox.geojson.utils.GeoJsonUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Adapter to read and write coordinates for Point class.
+ *
+ * @since 4.1.0
```
Comment (LukasPaczos) [inline services-geojson/src/main/java/com/mapbox/geojson/gson/BoundingBoxTypeAdapter.java]: This should be `4.4.0`.

File: services-geojson/src/main/java/com/mapbox/geojson/gson/BoundingBoxTypeAdapter.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import com.mapbox.geojson.exception.GeoJsonException;
+import com.mapbox.geojson.shifter.CoordinateShifterManager;
+import com.mapbox.geojson.utils.GeoJsonUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Adapter to read and write coordinates for Point class.
```
Comment (LukasPaczos) [inline services-geojson/src/main/java/com/mapbox/geojson/gson/BoundingBoxTypeAdapter.java]: This should mention `BoundingBox` class.

File: services-geojson/src/main/java/com/mapbox/geojson/BaseGeometryTypeAdapter.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+  BaseGeometryTypeAdapter(Gson gson, TypeAdapter<T> coordinatesAdapter) {
+    this.gson = gson;
+    this.coordinatesAdapter = coordinatesAdapter;
+    this.boundingBoxAdapter = new BoundingBoxTypeAdapter();
+  }
+
+  @SuppressWarnings("unchecked")
+  public void writeCoordinateContainer(JsonWriter jsonWriter, CoordinateContainer<T> object)
+          throws IOException {
```
[the remark this answers is not in the document]
Comment (osana) [reply services-geojson/src/main/java/com/mapbox/geojson/BaseGeometryTypeAdapter.java]: not sure what you mean.
throws IOException does not fit on the line above

**Pull request merged**